app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -29,49 +27,70 @@ import org.mozilla.fenix.GleanMetrics.RecentSyncedTabs
+    private val storage: SyncedTabsStorage,
+    private val accountManager: FxaAccountManager,
     private val coroutineScope: CoroutineScope,
-    accountManager: FxaAccountManager,
-    context: Context,
-    storage: SyncedTabsStorage,
-    lifecycleOwner: LifecycleOwner,
-) : SyncedTabsView, LifecycleAwareFeature {
-    private val syncedTabsFeature by lazy {
-        SyncedTabsFeature(
-            view = this,
-            context = context,
-            storage = storage,
-            accountManager = accountManager,
-            lifecycleOwner = lifecycleOwner,
-            onTabClicked = {}
-        )
-    }
-
-    override var listener: SyncedTabsView.Listener? = null
+) : LifecycleAwareFeature {
 
     private var syncStartId: GleanTimerId? = null
     private var lastSyncedTab: RecentSyncedTab? = null
 
-    override fun startLoading() {
+    override fun start() {
+        collectAccountUpdates()
+        collectStatusUpdates()
+    }
+
+    override fun stop() = Unit
+
+    private fun collectAccountUpdates() {
+        syncStore.flow()
+            .ifChanged { state ->
+                state.account != null
+            }.onEach { state ->
+                if (state.account != null) {
+                    dispatchLoading()
+                    // Sync tabs storage will fail to retrieve tabs aren't refreshed, as that action
+                    // is what populates the device constellation state
+                    accountManager.withConstellation { refreshDevices() }
+                    accountManager.syncNow(SyncReason.User, customEngineSubset = listOf(SyncEngine.Tabs))
+                }
+            }.launchIn(coroutineScope)
+    }
+
+    private fun collectStatusUpdates() {
+        syncStore.flow()
+            .ifChanged { state ->
+                state.status
+            }.onEach { state ->
+                when (state.status) {
+                    SyncStatus.Idle -> dispatchSyncedTabs()
+                    SyncStatus.Error -> onError()
+                    SyncStatus.LoggedOut -> appStore.dispatch(
+                        AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.None)
+                    )
+                    else -> Unit
+                }
+            }.launchIn(coroutineScope)
+    }
+
+    private fun dispatchLoading() {
         syncStartId?.let { RecentSyncedTabs.recentSyncedTabTimeToLoad.cancel(it) }
         syncStartId = RecentSyncedTabs.recentSyncedTabTimeToLoad.start()
         if (appStore.state.recentSyncedTabState == RecentSyncedTabState.None) {
             appStore.dispatch(AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.Loading))
         }
     }
 
-    override fun displaySyncedTabs(syncedTabs: List<SyncedDeviceTabs>) {
-        val syncedTab = syncedTabs
+    private suspend fun dispatchSyncedTabs() {
+        val syncedTab = storage.getSyncedDeviceTabs()
             .filterNot { it.device.isCurrentDevice || it.tabs.isEmpty() }
```

Comment:
you may not need the filter on `it.device.isCurrentDevice`. Based on my work in SyncedTabs in the tabs tray, I believe this happens during `storage.getSyncedDeviceTabs()`

Reply:
I think you're right, I may have been using multiple releases channels of Firefox on the same device and thought they were the same Sync device (since a nightly release and a debug release would show as different devices, for instance). 

That said, I don't think there's anything in the `getSyncedDeviceTabs` name that suggests it would guarantee that to _always_ be the case. I'm happy to change it, but I think in the interest of not being too introspective into the abstraction barrier that I'll otherwise leave it for now. It could also prevent future bugs if the underlying implementation were to change.

Reply:
I believe it's from the callstack of: `SyncedTabsStorage.getSyncedDeviceTabs()` -> `SyncedTabsStorage.syncClients()` -> `ConstellationState.otherDevices`

Either way, not a blocker for this by any means, and I agree that the name/callstack would ideally be more explicit about the filtering used.